### PR TITLE
Use the latest working IPFS hash from IdentityUpdated events

### PR DESCRIPTION
**Pull request by mikeshultz**
### Description:

Instead of trying to look for errors in IPFS hashes, just test them.  The only way to really be sure it works.

Closes #3583 
Ref #3581

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: packages/ipfs/src/index.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ async function post(gateway, json, rawHash) {
 //   return decrypted.data
 // }
 
-async function getTextFn(gateway, hashAsBytes) {
+async function getTextFn(gateway, hashAsBytes, timeoutMS) {
```
Comment (nick) [inline packages/ipfs/src/index.js]: should we set the default for this too?

File: packages/graphql/src/resolvers/IdentityEvents.js [hunk 2]
```diff
@@ -208,17 +209,29 @@ export function identity({ id, ipfsHash }) {
...
         return resolve(null)
       }
+    } else {
+      ipfsHashes.push(ipfsHash)
     }
 
-    const data = await originIpfs.get(contracts.ipfsGateway, ipfsHash)
+    // Go through each hash until we get valid data
+    let data
+    for (const hash of ipfsHashes) {
```
Comment (nick) [inline packages/graphql/src/resolvers/IdentityEvents.js]: Maybe set a max of 3 to avoid too many lookups?

File: packages/ipfs/src/index.js (hunk 1 above)
Comment (mikeshultz) [reply packages/ipfs/src/index.js]: There aren't any direct consumers of this function according to my grepping. Just internal in `@origin/ipfs`.

File: packages/graphql/src/resolvers/IdentityEvents.js (hunk 2 above)
Comment (mikeshultz) [reply packages/graphql/src/resolvers/IdentityEvents.js]: sg

**Comment by mikeshultz** [comment]
Fun fact, I think this accidentally fixes a potential DoS with overlarge files as well.

**Pull request merged**